lib/usd/utils/util.h

```diff
@@ -53,6 +53,10 @@ namespace MayaUsdUtils {
     MAYA_USD_UTILS_PUBLIC
     bool hasSpecs(const UsdPrim&);
 
+    //! Returns the layer in composition arc where HasSpecs is set to true
+    MAYA_USD_UTILS_PUBLIC
+    std::vector<SdfLayerHandle> layerInCompositionArcsWithSpec(const UsdPrim& prim);
```

Comment:
@ppt-adsk added this new utility function since the new design for the first restriction rule requires the layer's name to be displayed in the message.

Please see the new string for first rule:

`Error Message: Cannot [reparent/rename] [prim_name]. It does not make any contributions in the current layer because its specs are in an external composition arc. Please open [filename.usd] to make direct edits.`